src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,163 @@
+                declaredType = declaration.Type;
+
+                shouldAnalyze = ShouldAnalyzeVariableDeclaration(declaration, context.SemanticModel, context.CancellationToken);
+
+                if (shouldAnalyze)
+                {
+                    state = State.Generate(declarationStatement, context.SemanticModel, optionSet, isVariableDeclarationContext: true, cancellationToken: context.CancellationToken);
+                    shouldAnalyze = IsStylePreferred(declaration, context.SemanticModel, optionSet, state, context.CancellationToken);
+                }
+            }
+            else if (declarationStatement.IsKind(SyntaxKind.ForEachStatement))
+            {
+                var declaration = (ForEachStatementSyntax)declarationStatement;
+                declaredType = declaration.Type;
+
+                state = State.Generate(declarationStatement, context.SemanticModel, optionSet, isVariableDeclarationContext: false, cancellationToken: context.CancellationToken);
+                shouldAnalyze = IsStylePreferred(declaration, context.SemanticModel, optionSet, state, context.CancellationToken);
+            }
+            else
+            {
+                Debug.Assert(false, $"called in for unregistered node kind {declarationStatement.Kind().ToString()}");
+                return;
+            }
+
+            if (shouldAnalyze)
+            {
+                Debug.Assert(state != null, "analyzing a declaration and state is null.");
+
+                TextSpan diagnosticSpan;
+
+                if (TryAnalyzeVariableDeclaration(declaredType, context.SemanticModel, optionSet, context.CancellationToken, out diagnosticSpan))
+                {
+                    var descriptor = _severityToDescriptorMap[state.GetDiagnosticSeverityPreference()];
+                    context.ReportDiagnostic(CreateDiagnostic(descriptor.Value, declarationStatement, diagnosticSpan));
+                }
+            }
+        }
+
+        private Diagnostic CreateDiagnostic(DiagnosticDescriptor descriptor, SyntaxNode declaration, TextSpan diagnosticSpan) =>
+            Diagnostic.Create(descriptor, declaration.SyntaxTree.GetLocation(diagnosticSpan));
+
+        private bool ShouldAnalyzeVariableDeclaration(VariableDeclarationSyntax variableDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            // var is applicable only for local variables.
+            if (variableDeclaration.Parent.IsKind(SyntaxKind.FieldDeclaration) ||
+                variableDeclaration.Parent.IsKind(SyntaxKind.EventFieldDeclaration))
+            {
+                return false;
+            }
+
+            // implicitly typed variables cannot have multiple declarators and
+            // must have an initializer.
+            if (variableDeclaration.Variables.Count > 1 ||
+                !variableDeclaration.Variables.Single().Initializer.IsKind(SyntaxKind.EqualsValueClause))
```

Comment:
Note that this will throw if there are 0 variables due to the call to `Single(...)`. It doesn't look like that was your intent.